internal/broker/instance_create.go

```diff
@@ -412,6 +415,17 @@ func supportsAdditionalWorkerNodePools(planID string) bool {
 	return false
 }
 
+func areNamesUnique(pools []pkg.AdditionalWorkerNodePool) bool {
```

Comment:
You can manage without bool. :)

1. Count
2. Make set i.e. map[string]struct{}
3. Insert elements
4. Count elements in the set
5. compare checks in 1. and 4.

Reply:
However your solution is faster since we do not need to process all elements (on average).